lib/backend/k8s/conversion/conversion.go

```diff
@@ -51,10 +52,12 @@ func VethNameForWorkload(workload string) string {
 	return fmt.Sprintf("cali%s", hex.EncodeToString(h.Sum(nil))[:11])
 }
 
-// ParseWorkloadID extracts the Namespace and Pod name from the given workload ID.
-func (c Converter) ParseWorkloadID(workloadID string) (string, string) {
-	splits := strings.SplitN(workloadID, ".", 2)
-	return splits[0], splits[1]
+// ParseWorkloadName extracts the Node name, Orchestrator, Pod name and endpoint from the
+// given WorkloadEndpoint Resource name.
+// The expected format for k8s is <node>-k8s-<pod>-<endpoint>
+func (c Converter) ParseWorkloadName(workloadName string) (string, string, string, string) {
+	parts := names.ExtractDashSeparatedParms(workloadName, 4)
+	return parts[0], parts[1], parts[2], parts[3]
```

Comment:
CNI actually does a "prefix" query to find the current endpoint in a workload.

In this case it's possible to get only 3 parts to the name.  In any case this feels like it should error if it can't deconstruct the name otherwise won't it panic?

Reply:
i.e. For a List(), if the prefix flag is set in the ListOptions, then we expect 3 parameters.  For all other actions - we expect all 4.

Reply:
I'm wondering if we should beef up the WorkloadEndpointIdentifiers thing in /lib/names so that the values can be assigned from the name - and that way you use that and pass around a WorkloadEndpointIdentifiers struct rather than returning multiple values.

Reply:
Yup. I was thinking along similar lines. It becomes unwieldy to handle the error otherwise.